server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1346,6 +1346,15 @@ public String toString() {
         Property.InternalIndex
     );
 
+    public static final String INDEX_IGNORE_DEPRECATION_WARNING_FOR_VERSION_KEY = "index.ignore_deprecation_warning_for_version";
+    public static final Setting<IndexVersion> IGNORE_INDEX_DEPRECATION_WARNING_FOR_VERSION_SETTING = Setting.versionIdSetting(
+        INDEX_IGNORE_DEPRECATION_WARNING_FOR_VERSION_KEY,
+        IndexVersions.ZERO,
+        IndexVersion::fromId,
+        Property.IndexScope,
+        Property.Dynamic
```

Comment:
I thin it might make sense to add the nonCopyableOnResize property so this get's removed if the index is cloned